examples/IrisClassificationTorchScript/iris_datamodule.py

```diff
@@ -11,21 +12,25 @@ def __init__(self):
         super().__init__()
 
     def prepare_data(self):
-        iris = load_iris()
-        df = iris.data
-        self.columns = iris.feature_names
-        target = iris["target"]
-        data = torch.Tensor(df).float()
-        labels = torch.Tensor(target).long()
-        data_set = TensorDataset(data, labels)
-        return data_set
+        """
+        Implementation of abstract class
+        """
 
     def setup(self, stage=None):
 
         # Assign train/val datasets for use in dataloaders
         if stage == "fit" or stage is None:
-            iris_full = self.prepare_data()
-            self.train_set, self.val_set = random_split(iris_full, [130, 20])
+            iris = load_iris()
+            df = iris.data
+            target = iris["target"]
+
+            data = torch.Tensor(df).float()
+            labels = torch.Tensor(target).long()
+            RANDOM_SEED = 42
+            seed_everything(RANDOM_SEED)
+
+            data_set = TensorDataset(data, labels)
```

Comment:
What is the reason for moving this from "prepare_data" to "setup" method?

Reply:
There is a change in behaviour from PTL `1.3.x` and ` 1.4.x`. 

`prepare_data` method does not return anything. Since the function calls are controlled by PTL DataModule class, prepare_data is getting called and the dataset is not passed on to the setup (as prepare_data doesnt return anything).

However, in the older version it is not the case. The behaviour can be observed in screenshots

PTL 1.3.x:

```
Returning from prepare data:  <torch.utils.data.dataset.TensorDataset object at 0x7f41043eb6a0>
Received in setup :  <torch.utils.data.dataset.TensorDataset object at 0x7f41043eb6a0>

```
PTL 1.4.x:
```
Returning from prepare data:  <torch.utils.data.dataset.TensorDataset object at 0x7fa58f2d1af0>
Received in setup :  None

```
 
![image](https://user-images.githubusercontent.com/63862647/128127562-71023b96-e27f-42ab-a7b6-067b15d35ded.png)
![image](https://user-images.githubusercontent.com/63862647/128127626-e1c05b72-774d-494c-984c-9d54591a863a.png)


Since the data preparation is simple in iris classification, @KasirajanA  has moved the logic to setup (instead of introducing another class variable).